WordPress/gutenberg
# Add ToggleMultipleGroupControl component

Extracted from #44168 (where a high-level review was completed first)

## What?

Add a ToggleMultipleGroupControl component.

## Why?

We need a component that looks like ToggleGroupControl, but allows multiple selection. Splitting this variant into a thin wrapper component keeps the API simpler. (See #44168 for details)

## How?

Exports two new components — ToggleMultipleGroupControl and ToggleMultipleGroupControlOptionIcon — that are just thin wrappers around the original ToggleGroupControl components. Conditional logic is handled internally, and the API surface is kept simple.

From a DX standpoint, the main difference is that the consumer needs to manage the `isPressed` states on their own for each of the option buttons via an `onClick` handler. This seems like more work, but it's actually a simpler DX compared to having the main ToggleMultipleGroupControl component handle all the states and centrally and trying to read them through a single `onChange`.

## Testing Instructions

1. `npm run storybook:dev` and check the docs/stories for ToggleMultipleGroupControl.
2. Check that the documentation makes sense.

✅ Unit tests pass

## Screenshots or screencast 

<img width="131" alt="ToggleMultipleGroupControl" src="https://user-images.githubusercontent.com/555336/196783707-fd42f36f-e4ab-434a-8adf-93095fc67525.png">
